arcgisUM.py

```diff
@@ -0,0 +1,206 @@
+# Wrapper around calls to arcgis.  Helps with testing and future changes.
+
+import datetime
+import logging
+
+logger = logging.getLogger(__name__)
+
+from arcgis.gis import GIS
+
+import dateutil.tz
+
+import config
+
+# secrets really is used during (import to change sensitive properties).
+import secrets  # @UnusedImport
+
+import util
+
+##### Improved code tracebacks for exceptions
+import traceback
+
+def handleError(self, record):  # @UnusedVariable
+    traceback.print_stack()
+    
+logging.Handler.handleError = handleError
+#####
+
+TIMEZONE_UTC = dateutil.tz.tzutc()
+RUN_START_TIME = datetime.datetime.now(tz=TIMEZONE_UTC)
+RUN_START_TIME_FORMATTED = RUN_START_TIME.strftime('%Y%m%d%H%M%S')
+
+# Hold parsed options
+options = None
+
+# TODO: required in this module?
+courseLogHandlers = dict()
+courseLoggers = dict()
```

Comment:
you can also use `courseLoggers={}` to initialize dictionary, is it a preference? or does use one over the other has any benefits?

Reply:
This is old code and appears to work, so I don't plan to change it.